lib/handleError.js

```diff
@@ -13,8 +13,8 @@ module.exports = {
     }
     if (get(response, 'body.errorCode')) {
       throw new Error(translateErrorCode(response.body.errorCode));
-    } else {
-      throw new Error(translateErrorCode('ERR_NO_BACKEND'));
+    } else if (get(response, 'error')) {
```

Comment:
I see that it's only possible to be used as 
```res.status(404).send('UPGRADE REQUIRED')``` to fill `response.error`, however the contents may be `<!DOCTYPE>...` - a response from a misconfigured server which will dump the HTML into the console.

Reply:
@vogre  I don't believe this is the case. We construct the error object in `deploy` and `remove` and explicitly set the `error` field. In the case of us sending something misconfigured I believe that `request` will surface that via normal error handling.

```
  try {
    const deployURLBase = `https://${endpoint}`;
    const codeDigest = await deployCode(deployURLBase, apiKey, tarPath);
    const confWithDigest = Object.assign({}, funcConf, { codeDigest });
    const confDeployResp = await deployConf(deployURLBase, apiKey,
      funcName, confWithDigest);
    return { status: confDeployResp.statusCode, body: confDeployResp.body };
  } catch (err) {
    if (err instanceof HTTPError) {
      return { status: err.response.statusCode, body: err.response.body };
    }
    // NOTE: This 'err' returned in the error field is in NodeJS error format
    return { error: err };
  }
```